stablehlo/reference/Ops.cpp

```diff
@@ -224,6 +225,47 @@ Tensor evalTransposeOp(const Tensor &operand, ArrayRef<int64_t> permutation,
   return result;
 }
 
+// ***********************************************************************
+// This is an simplified implementation of while op semantics. The
+// simplification is based on iterating the loop `dummy_limit` number of times
+// if the loop condition is `true`. The simplifiction is done as compare op is
+// not supported. To be corrected and improved as part of #967 and #992
+// respectively.
+// ***********************************************************************
+SmallVector<Tensor> evalWhileOp(ArrayRef<Tensor> runtimeInputs, Region &cond,
+                                Region &body, const InterpreterScope &scope) {
+  SmallVector<Tensor> runtimeResults(runtimeInputs);
+  bool isIsolatedFromAbove =
+      cond.getParentOp()->hasTrait<OpTrait::IsIsolatedFromAbove>();
```

Comment:
Don't we know this statically?

Re: "Please note that these checks make the code adjust to any future changes in the behavior of the op w.r.t its visibility to outer scope". I would imagine that we'll have this covered when/if we actually introduce these changes?

Reply:
makes sense.